digdag-core/src/main/java/io/digdag/core/agent/RequireOperatorFactory.java

```diff
@@ -64,46 +68,109 @@ private RequireOperator(OperatorContext context, TaskCallbackApi callback)
         public TaskResult runTask()
         {
             Config config = request.getConfig();
+            Config localConfig = request.getLocalConfig();
+            Config lastStateParams = request.getLastStateParams();
+
             String workflowName = config.get("_command", String.class);
             Instant instant = config.get("session_time", Instant.class);
             boolean ignoreFailure = config.get("ignore_failure", boolean.class, false);
-            Optional<String> retryAttemptName = config.getOptional("retry_attempt_name", String.class);
+            OptionRerunOn rerunOn = OptionRerunOn.of(config.get("rerun_on", String.class, "none"));
+
+            //retry_attempt_name is set from local config only.
+            Optional<String> retryAttemptName = localConfig.getOptional("retry_attempt_name", String.class);
+            if (lastStateParams.has("rerun_on_retry_attempt_name")) { // set for rerun_on parameter.
+                retryAttemptName = lastStateParams.getOptional("rerun_on_retry_attempt_name", String.class);
+            }
+
             Config overrideParams = config.getNestedOrGetEmpty("params");
+
+            Optional<StoredSessionAttempt> attempt = Optional.absent();
+            Optional<SessionAttemptConflictException> sessionAttemptConflictException = Optional.absent();
+            Optional<ProjectIdentifier> projectIdentifier = Optional.absent();
+            /**
+             *  First of all, try to start attempt by startSession()
+             *  If no attempt exists (no conflict), it return new StoredSessionAttempt.
+             *    - set state param "require_kicked" to true.
+             *    - task is retried to wait for done by nextPolling.
+             *  If something errors happen, it will throw following exceptions.
+             *    - ResourceNotFoundException ... workflow ,project name(or id) are wrong. -> processed as deterministic error
+             *    - ResourceLimitExceededException ... this exception should be deterministic error
+             *    - SessionAttemptConflictException ... this is not error.
+             *      - If the conflict attempt is still running, wait for until done by nextPolling.
+             *      - If done, check the state param "require_kicked" and whether the attempt is kicked by this require> or not.
```

Comment:
What about the case the conflict attempt failed like `If failed, xxx`?

Reply:
Do you mean, if the attempt which is kicked by require> failed,  what happen?
In this case the task will fail as usual.